ckanext/dia/model.py

```diff
@@ -66,10 +66,12 @@ def define_table():
 
 def name_and_type_unique(name, context):
     type = context.get('type')
+    model_id = context.get('id', 0)
     result = MintedURI.Session.query(MintedURI)\
         .filter(func.lower(MintedURI.name) == name.lower())\
         .filter(func.lower(MintedURI.type) == type.lower())\
         .filter(MintedURI.superseded_by == None)\
+        .filter(MintedURI.id != int(model_id))\
```

Comment:
This means that when editing an existing instance, we only validate against other instances with the same name and type, not the one being edited.